lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+/// precision.
+const PRECISION_ADJUSTMENT_STEP_BASE: f32 = 10.0;
+/// The maximum number of digits after the decimal point to be displayed when showing the
+/// component's value.
+const VALUE_DISPLAY_MAX_DIGITS_AFTER_DECIMAL: usize = 8;
+
+
+
+// ========================
+// === Slider component ===
+// ========================
+
+/// A slider UI component that allows adjusting a value through mouse interaction. Dragging the
+/// slider in a horizontal direction changes the value, limited to a range between `min_value` and
+/// `max_value`. The selected value is displayed, and a track fills the slider proportional to the
+/// value within the specified range. Dragging the slider in a vertical direction adjusts the
+/// precision of the slider. The precision affects the increments by which the value changes when
+/// the mouse is moved.
+#[derive(Debug, Deref)]
+pub struct Slider {
+    /// Public FRP api of the component.
+    #[deref]
+    pub frp: Frp,
+    model:   Rc<Model>,
+    /// Reference to the application the component belongs to. Generally required for implementing
+    /// `application::View` and initialising the `Model` and `Frp` and thus provided by the
+    /// component.
+    pub app: Application,
+}
+
+impl Slider {
+    /// Construct a new slider component.
+    pub fn new(app: &Application) -> Self {
+        let model = Rc::new(Model::new(app));
+        let app = app.clone_ref();
+        let frp = Frp::new();
+        Self { frp, model, app }.init()
+    }
+
+    fn init(self) -> Self {
+        let network = self.frp.network();
+        let input = &self.frp.input;
+        let output = &self.frp.private.output;
+        let model = &self.model;
+        let scene = &self.app.display.default_scene;
+        let mouse = &scene.mouse.frp;
+        let keyboard = &scene.keyboard.frp;
+        let track_pos_anim = Animation::new_non_init(network);
+        let background_color_anim = color::Animation::new(network);
+        let track_color_anim = color::Animation::new(network);
+        let value_text_color_anim = color::Animation::new(network);
+        let label_color_anim = color::Animation::new(network);
+
+        frp::extend! { network
+
+            // === User input ===
+
+            background_click <- model.background.events.mouse_down_primary.constant(());
+            background_release <- model.background.events.mouse_release_primary.constant(());
+            background_drag <- bool(&background_release,&background_click);
+            track_click <- model.track.events.mouse_down_primary.constant(());
+            track_release <- model.track.events.mouse_release_primary.constant(());
+            track_drag <- bool(&track_release,&track_click);
+            component_click <- any2(&background_click,&track_click);
+            component_click <- component_click.gate_not(&input.set_slider_disabled);
+            component_drag <- any2(&background_drag,&track_drag);
```

Comment:
this is pretty complex. Instead of checking clicking on track and background, just set track to ignore mouse events and pass them trough it. There is a setting for that when defining the shape. I dont remember the name, but its something alongtheu line of `disable_pointer_events`. Then you can watch events on bg only.